packages/progress-circle/src/ProgressCircle.ts

```diff
@@ -101,6 +114,12 @@ export class ProgressCircle extends SizedMixin(SpectrumElement, {
         `;
     }
 
+    private onSlotChange(): void {
+        if (!this.label && this.slotHasContent) {
+            this.label = this.slotEl.assignedNodes()[0].textContent || '';
+        }
```

Comment:
This pattern can effect a consumers ability to leverage the `label` attribute/property themselves, maybe we don't use it as an interchange. Something like:
```
if (this.label) return;
const textContent = this.slotEl.assignedNodes()[0].textContent;
if (textContent) {
   // set attribute
} else {
  // remove attribute
}
```
Also, you likely want to do a `reduce` across all the assigned nodes to acquire this value.